app/ts/types.ts

```diff
@@ -1,34 +1,24 @@
-import { BigNumber, Contract, ethers } from 'ethers'
-
-export type Web3Provider = ethers.providers.Web3Provider
-export type ExternalProvider = ethers.providers.ExternalProvider
-export type TransactionResponse = ethers.providers.TransactionResponse
-export type Network = ethers.providers.Network
-export type TransactionReceipt = ethers.providers.TransactionReceipt
+import { Contract, Eip1193Provider, TransactionResponse } from 'ethers'
 
 declare global {
 	interface Window {
-		ethereum?: ExternalProvider
+		ethereum?: Eip1193Provider
 	}
 }
 
 export type HexString = `0x${string}`
 
 export interface ERC20 extends Contract {
-	balanceOf(address: string): Promise<BigNumber>
-	totalSupply(): Promise<BigNumber>
-	transfer(to: string, value: BigNumber): Promise<TransactionResponse>
-	transferFrom(from: string, to: string, value: BigNumber): Promise<boolean>
-	approve(spender: string, value: BigNumber): Promise<boolean>
-	allowance(owner: string, spender: string): Promise<BigNumber>
+	balanceOf(address: string): Promise<BigInt>
+	totalSupply(): Promise<BigInt>
+	transfer(to: string, value: BigInt): Promise<TransactionResponse>
+	transferFrom(from: string, to: string, value: BigInt): Promise<boolean>
+	approve(spender: string, value: BigInt): Promise<boolean>
+	allowance(owner: string, spender: string): Promise<BigInt>
```

Comment:
For BigInt types use ``bigint``, use the ``BigInt`` for casting to bigints like `BigInt(123)` or `Bigint('0xabc')`

Reply:
sorry these were old codes i left there. removed them now as I'm not using the type for now